File: backend/python/models/lesson_model.py
from db.mongodb import get_db
from bson import ObjectId
from datetime import datetime

def complete_lesson(user_id, lesson_id):
    db = get_db()
    existing = db.completed_lessons.find_one({'user_id': ObjectId(user_id), 'lesson_id': lesson_id})
    if existing:
        return None
    
    completed = {
        'user_id': ObjectId(user_id),
        'lesson_id': lesson_id,
        'completed_at': datetime.utcnow()
    }
    result = db.completed_lessons.insert_one(completed)
    completed['_id'] = result.inserted_id
    return completed

def get_completed_lessons(user_id):
    db = get_db()
    lessons = list(db.completed_lessons.find({'user_id': ObjectId(user_id)}))
    return lessons

File: backend/node/models/lessonModel.js
const { getDB } = require('../db/mongodb');
const { ObjectId } = require('mongodb');

const getCompletedLessons = async (userId) => {
  const db = getDB();
  if (!db) return [];
  
  const lessons = await db.collection('completed_lessons')
    .find({ user_id: new ObjectId(userId) })
    .toArray();
  return lessons;
};

const completeLesson = async (userId, lessonId) => {
  const db = getDB();
  if (!db) return false;
  
  const existing = await db.collection('completed_lessons').findOne({
    user_id: new ObjectId(userId),
    lesson_id: lessonId
  });
  
  if (existing) return false;
  
  await db.collection('completed_lessons').insertOne({
    user_id: new ObjectId(userId),
    lesson_id: lessonId,
    completed_at: new Date()
  });
  
  return true;
};

module.exports = { getCompletedLessons, completeLesson };
